### PR TITLE
Api error handling

### PR DESCRIPTION
**Pull request by da-saman**

### CODE REVIEW

File: src/components/shared/KInfoMessage/KInfoMessage.style.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export const KInfoMessageStyled = styled.div`
+    border-radius: 8px;
+    padding: 10px;
+    position: fixed;
+    bottom: 10px;
+    background-color: #FF6C47;
+    color: white;
+    right: 10px;
+    animation: positionAnimate 0.05s ease-in-out;
+    z-index: 10;
```
Comment (salammohamadi) [inline src/components/shared/KInfoMessage/KInfoMessage.style.tsx:12]: use much bigger number for z-index
such as   ===>  z-index: 10000000000;

when any modal is open if encountered an error it won't be shown above it

File: src/api/core/request.ts [hunk 2]
```diff
@@ -160,19 +162,17 @@ function catchErrors(options: ApiRequestOptions, result: ApiResult): void {
     401: 'Unauthorized',
     403: 'Forbidden',
     404: 'Not Found',
+    422: 'Unprocessable Entity.',
```
Comment (salammohamadi) [inline src/api/core/request.ts:165]: options: ApiRequestOptions ===> first argument not have been used ?

File: src/assets/icons/CloseCircle.tsx [hunk 3]
```diff
@@ -3,13 +3,13 @@ import React from 'react';
 import { IconProps } from './iconType';
 
 export const CloseCircle = (props: IconProps) => {
-  const { width, height } = props;
+  const { width, height, color } = props;
```
Comment (salammohamadi) [inline src/assets/icons/CloseCircle.tsx:6]: default value seems cleaner syntax
const { width, height, color='455A64' } = props;

stroke={color || '#455A64'} =>> change it to stroke={color}

File: src/components/shared/KInfoMessage/KInfoMessage.style.tsx (hunk 1 above)
Comment (da-saman) [reply src/components/shared/KInfoMessage/KInfoMessage.style.tsx:12]: fixed, set to 1001

File: src/api/core/request.ts (hunk 2 above)
Comment (da-saman) [reply src/api/core/request.ts:165]: removed

File: src/assets/icons/CloseCircle.tsx (hunk 3 above)
Comment (da-saman) [reply src/assets/icons/CloseCircle.tsx:6]: thanks, done

File: src/components/shared/KInfoMessage/KInfoMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      dispatch(setApiErrors([]));
+    }, 4000);
+
+    return () => {
+      window.clearTimeout(timeOut);
+    };
+  }, []);
+  return (
+    <KInfoMessageStyled>
+      <>
```
[the remark this answers is not in the document]
Comment (da-saman) [reply src/components/shared/KInfoMessage/KInfoMessage.tsx:27]: removed

**Review by salammohamadi — APPROVED** [review]

**Pull request merged**